Lib/profiling/sampling/live_collector/widgets.py

```diff
@@ -0,0 +1,963 @@
+                line, col, f"{funcs_shown:>5}", self.colors["magenta"]
+            )
+            col += 5
+            self.add_str(line, col, " shown", curses.A_NORMAL)
+        return line + 1
+
+    def draw_top_functions(self, line, width, stats_list):
+        """Draw top N hottest functions."""
+        col = 0
+        self.add_str(
+            line,
+            col,
+            f"Top {TOP_FUNCTIONS_DISPLAY_COUNT}:     ",
+            curses.A_BOLD,
+        )
+        col += 11
+
+        top_by_samples = sorted(
+            stats_list, key=lambda x: x["direct_calls"], reverse=True
+        )
+        emojis = ["🥇", "🥈", "🥉"]
+        medal_colors = [
+            self.colors["red"],
+            self.colors["yellow"],
+            self.colors["green"],
+        ]
+
+        displayed = 0
+        for func_data in top_by_samples:
+            if displayed >= TOP_FUNCTIONS_DISPLAY_COUNT:
+                break
+            if col >= width - 20:
+                break
+            if func_data["direct_calls"] == 0:
+                continue
+
+            func_name = func_data["func"][2]
+            func_pct = (
+                func_data["direct_calls"]
+                / max(1, self.collector.total_samples)
+            ) * 100
+
+            # Medal emoji
+            if col + 3 < width - 15:
+                self.add_str(
+                    line, col, emojis[displayed] + " ", medal_colors[displayed]
+                )
+                col += 3
+
+            # Function name (truncate to fit)
+            available_for_name = width - col - 15
+            max_name_len = min(25, max(5, available_for_name))
+            if len(func_name) > max_name_len:
+                func_name = func_name[: max_name_len - 3] + "..."
+
+            if col + len(func_name) < width - 10:
+                self.add_str(line, col, func_name, medal_colors[displayed])
+                col += len(func_name)
+
+                pct_str = (
+                    f" ({func_pct:.1f}%)"
+                    if func_pct >= 0.1
+                    else f" ({func_data['direct_calls']})"
+                )
+                self.add_str(line, col, pct_str, curses.A_DIM)
+                col += len(pct_str)
+
+                displayed += 1
+
+                if displayed < 3 and col < width - 30:
+                    self.add_str(line, col, " │ ", curses.A_DIM)
+                    col += 3
+
+        if displayed == 0 and col < width - 25:
+            self.add_str(line, col, "(collecting samples...)", curses.A_DIM)
+
+        return line + 1
+
+    def draw_finished_banner(self, line, width):
+        """Draw a prominent banner when profiling is finished."""
+        A_REVERSE = self.display.get_attr("A_REVERSE")
+        A_BOLD = self.display.get_attr("A_BOLD")
+
+        # Add blank line for separation
+        line += 1
+
+        # Create the banner message
+        message = "  ✓ PROFILING COMPLETE - Final Results Below - Press 'q' to Quit  "
```

Comment:
One thing we could do here is have a better call to action when the profiling is complete. It would be nice if we had a way to go from TUI -> export data after the live session. I could just as well run this again with different flags but it'd be nice to have a really simple flow from live viewing to data export.